### PR TITLE
fix: Depreciate asset on the last day of the month if its depr start date falls on the last day of its month

### PR DESCRIPTION
**Pull request by GangaManoj**
## Issue

If an Asset that depreciates monthly starts depreciating on, say, Feb 28, then it would depreciate next on March 28, then April 28 and so on. However, customers usually intend to have the asset depreciate on the last day of each month in this case- i.e. Feb 28 should be followed by March 31 and April 30 instead.

## Fix

This PR checks if the `depreciation_start_date` falls on the last day of its month, and if so, modify the schedule dates so that they'd each fall on the last day of their respective months.

Screenshot 2022-06-09 at 2 38 31 AM

### CODE REVIEW

File: erpnext/assets/doctype/asset/asset.py [hunk 1]
```diff
@@ -405,6 +409,14 @@ def make_depreciation_schedule(self, date_of_sale):
...
 						)
 
+	def is_last_day_of_the_month(self, depreciation_start_date):
+		last_day_of_the_month = get_last_day(depreciation_start_date)
+
+		if getdate(last_day_of_the_month) == getdate(depreciation_start_date):
+			return True
+		else:
+			return False
+
```
Comment (marination) [inline erpnext/assets/doctype/asset/asset.py]:
```suggestion
		return getdate(last_day_of_the_month) == getdate(depreciation_start_date)
```

File: erpnext/assets/doctype/asset/test_asset.py [hunk 2]
```diff
@@ -1261,6 +1261,32 @@ def test_asset_cost_center(self):
...
+			"2020-02-28",
+			"2020-03-31",
+			"2020-04-30",
+			"2020-05-31",
+			"2020-06-30",
+			"2020-07-15",
+		]
+
+		for i, schedule in enumerate(asset.schedules):
+			self.assertEqual(expected_dates[i], schedule.schedule_date)
```
Comment (marination) [inline erpnext/assets/doctype/asset/test_asset.py]: Failing CI
```suggestion
			self.assertEqual(getdate(expected_dates[i]), schedule.schedule_date)
```

**Comment by GangaManoj** [comment]
Fix included in https://github.com/frappe/erpnext/pull/31429, thereby closing this.

**Comment by GangaManoj** [comment]
Fix included in https://github.com/frappe/erpnext/pull/31429, therefore closing this.

**Pull request closed without merging**